app/controllers/user_controller.rb

```diff
@@ -53,7 +56,7 @@ def user_params
   end
 
   def check_authorized_access
-    raise CanCan::AccessDenied.new("Unauthorized access!", :read, User) unless ((@user.admin_id == current_user.id) && (@user.id != current_user.id) && (@user.role != "Admin"))
+    raise CanCan::AccessDenied.new("Unauthorized access!", :read, User) unless ((@user.admin_id == current_user.id) && (@user.id != current_user.id) && (@user.role != "admin"))
```

Comment:
There will be lots of admin user in database so checking only admin will not be sufficient. Admin user of company can only access his company's users.